src/api/attachments.tsx

```diff
@@ -19,8 +26,52 @@ export const usePostAttachmentMetadata = (): UseMutationResult<
   AxiosError,
   AttachmentPostMetadata
 > => {
+  const queryClient = useQueryClient();
   return useMutation({
     mutationFn: (attachmentMetadata: AttachmentPostMetadata) =>
       postAttachmentMetadata(attachmentMetadata),
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: ['Attachments'],
+      });
+    },
+  });
```

Comment:
Currently, this invalidates all `Attachments` queries after a successful post request.  

This isn't ideal for our use case because we use Uppy, which allows multiple simultaneous uploads. Invalidating all queries every time an upload completes is excessive.  

Instead, we should only invalidate the query for the attachments related to the current entity (the page we are on). There's an example of this in `UploadImagesDialog`.

Reply:
@rowan04